pkg/controllers/externaloidc/externaloidc_controller_test.go

```diff
@@ -0,0 +1,964 @@
+				str = strings.ReplaceAll(str, "another-aud", "yet-another-aud")
+				return str
+			}(),
+			expectEvents: true,
+			expectError:  false,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			objects := []runtime.Object{}
+
+			authIndexer := cache.NewIndexer(func(obj interface{}) (string, error) {
+				return "cluster", nil
+			}, cache.Indexers{})
+
+			if tt.configMapIndexer == nil {
+				tt.configMapIndexer = cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+			}
+
+			if tt.auth != nil {
+				authIndexer.Add(tt.auth)
+			}
+
+			if tt.caBundleConfigMap != nil {
+				tt.configMapIndexer.Add(&baseCABundleConfigMap)
+			}
+
+			if tt.existingAuthConfigCM != nil {
+				tt.configMapIndexer.Add(tt.existingAuthConfigCM)
+				objects = append(objects, tt.existingAuthConfigCM)
+			}
+
+			cs := fake.NewSimpleClientset(objects...)
+			if tt.cmApplyReaction != nil {
+				// fake client's Apply() is a patch under the hood
+				cs.PrependReactor("patch", "configmaps", tt.cmApplyReaction)
+			}
+
+			if tt.cmDeleteReaction != nil {
+				cs.PrependReactor("delete", "configmaps", tt.cmDeleteReaction)
+			}
+
+			c := externalOIDCController{
+				configMaps:      cs.CoreV1(),
+				authLister:      configv1listers.NewAuthenticationLister(authIndexer),
+				configMapLister: corev1listers.NewConfigMapLister(tt.configMapIndexer),
+			}
+
+			eventRecorder := events.NewInMemoryRecorder("externaloidc-test")
+			err := c.sync(testCtx, factory.NewSyncContext("externaloidc-test-context", eventRecorder))
+			if tt.expectError && err == nil {
+				t.Errorf("expected error but didn't get any")
+			}
+
+			if !tt.expectError && err != nil {
+				t.Errorf("did not expect any error but got: %v", err)
+			}
+
+			if tt.expectEvents != (len(eventRecorder.Events()) > 0) {
+				t.Errorf("expected events: %v; got %v", tt.expectEvents, eventRecorder.Events())
+			}
+
+			if tt.auth == nil || err != nil {
+				// stop assertions here; the ones that follow are not relevant
+				return
+			}
+
+			cm, err := c.configMaps.ConfigMaps(managedNamespace).Get(testCtx, targetAuthConfigCMName, metav1.GetOptions{})
+			if len(tt.expectedAuthConfigJSON) == 0 && err == nil {
+				t.Errorf("expected auth configmap to be missing, but it was found")
+			} else if len(tt.expectedAuthConfigJSON) > 0 && errors.IsNotFound(err) {
+				t.Errorf("expected auth configmap to exist but it was not found; error = %v", err)
+			}
+
+			if len(tt.expectedAuthConfigJSON) > 0 && tt.expectedAuthConfigJSON != cm.Data[authConfigDataKey] {
+				t.Errorf("got unexpected auth-config data: '%s'\nexpected: '%s'", cm.Data[authConfigDataKey], tt.expectedAuthConfigJSON)
+			}
+		})
+	}
+}
+
+func TestExternalOIDCController_generateAuthConfig(t *testing.T) {
+	testServer, err := createTestServer(baseCACert, baseCAPrivateKey, nil)
+	if err != nil {
+		t.Fatalf("could not create test server: %v", err)
+	}
+	defer testServer.Close()
+	testServer.StartTLS()
```

Comment:
Do we need this test server for testing the auth config generation?

Reply:
Right, that's a leftover -- CA validation used to be inside `generateAuthConfig()` :+1: